src/Compilers/CSharp/Portable/Lowering/LambdaRewriter/SynthesizedClosureEnvironment.cs

```diff
@@ -25,6 +27,9 @@ internal sealed class SynthesizedClosureEnvironment : SynthesizedContainer, ISyn
         internal readonly FieldSymbol SingletonCache;
         internal readonly MethodSymbol StaticConstructor;
 
+        private ArrayBuilder<Symbol> _membersBuilder = ArrayBuilder<Symbol>.GetInstance();
```

Comment:
>private ArrayBuilder<Symbol> _membersBuilder = ArrayBuilder<Symbol>.GetInstance(); [](start = 8, length = 82)

It looks like we are leaking the builder from the pool, perhaps we shouldn't use a pool for this field.